app/models/apple/key.rb

```diff
@@ -17,5 +18,14 @@ def provider_id_is_valid
     def key_pem
       Base64.decode64(key_pem_b64)
     end
+
+    def correct_key_pem_format
+      header = "-----BEGIN"
+      footer = "-----END"
+      lines = key_pem.split("\n")
+      unless lines[0].include?(header) && lines[-1].include?(footer)
+        errors.add(:key_pem, "key pem does not have correct header or footer")
+      end
+    end
```

Comment:
Really cool to see. This validation goes beyond what is necessary to fix the original problem and starts to posit the question: "What if someone uploads a bad/corrupted keyfile?".  From a UI perspective, it's good to guard against bad input!

A couple thoughts:

1) The current approach is to blow up. But it's opaque to the user because it happens in a background job -- not good! The user would upload a bad key, then presumably create an episode with some audio, and a background job is kicked off where we [instantiate the key class](https://github.com/PRX/feeder.prx.org/blob/main/app/models/apple/api.rb#L60) and _then_ it blows up. So this validation better. We don't want the user to upload malformed keys.

2) The OpenSSL class [we are using for the ecdsa key](https://github.com/PRX/feeder.prx.org/blob/main/app/models/apple/api.rb#L60), apparently, does some kind of validation on the key. And this (in an engineering sense) is a _surface_ of potential bugs and problems. We take the (potentially corrupted) key and feed it into this black box.  But it seems reasonable to assume that the key format has to be *at least* consistent in order to build up this ECDSA representation in the openssl class, right?

Re: 2,  would be better to wrap the current approach with this validation? So instance of testing the correctness of the PEM format, we use `OpenSSL::PKey::EC.new(...)` to test the key material and rescue from any exceptions?

Reply:
q2: it seems like that is out of scope for what i'm trying to achieve? if i'm understanding the question, the suggestion is to test that the key that _comes out_ of the "black box" of the OpenSSL class method is formatted correctly. but it feels like that's not what this model is for? this `Apple::Key` model holds the information given to it from the user, which is what the `Apple::API` then uses, right?

but i don't think that's what i'm trying to validate here. i'm only validating that this key that _goes into_ the "black box" has the correct format. anything that happens beyond that point feels like it should belong to either `Apple::API` or wherever else that that OpenSSL method is used to create a key. this validation is only based off of [the PEM documentation as it currently exists](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail), and that's the only thing i'm trying to match to, which is why it's hard-coded in this way.